src/Data/String/NonEmpty/Internal.purs

```diff
@@ -13,7 +19,15 @@ import Prim.TypeError as TE
 import Unsafe.Coerce (unsafeCoerce)
 
 -- | A string that is known not to be empty.
-newtype NonEmptyString = NonEmptyString String
+newtype NonEmptyString =
+  -- | You can use this constructor to create a NonEmptyString that isn't
+  -- | non-empty, breaking the guarantee behind this newtype. It is
+  -- | provided as an escape hatch mainly for the `Data.NonEmpty.CodeUnits`
+  -- | and `Data.NonEmpty.CodePoints` modules. Usage of this constructor
+  -- | in your own code provides the same safety guarantees of
```

Comment:
This isn't quite true; the NonEmptyString constructor at least requires that you provide a string, whereas `unsafeCoerce` doesn't. That is, you can do `unsafeCoerce 3 :: NonEmptyString`, whereas you can't do `NonEmptyString 3 :: NonEmptyString`.

Reply:
Sorry, I missed this before.

Reply:
Lol. No worries. What are you suggestions for how to fix it?

Reply:
Can’t we just remove this sentence? The first one already mention that usage of this constructor breaks the newtype guarantees.

Reply:
Yeah, that sounds good to me.